src/main/scala/com/fulcrumgenomics/bam/ClipBam.scala

```diff
@@ -66,6 +67,7 @@ class ClipBam
   @arg(flag='r', doc="Reference sequence fasta file.") val ref: PathToFasta,
   @arg(flag='s', doc="Soft clip reads instead of hard clipping.") val softClip: Boolean = false,
   @arg(flag='a', doc="Automatically clip extended attributes that are the same length as bases.") val autoClipAttributes: Boolean = false,
+  @arg(flag='H', doc="Convert all soft-clipping, including existing soft-clipping, to hard-clipping (requires `--soft-clip=true`).") val convertSoftClipToHardClip: Boolean = false,
```

Comment:
This is super confusing.  What this means is that if I just want to run the tool to convert existing soft-clips to hard-clips I have to specify `--soft-clip=true --convert-soft-clip-to-hard-clip=true`, which seems unnecessary.

Secondly, it's also really confusing (and implies some knowledge of the ordering in the code) that if I want to do some clipping and also convert existing clips to hard clips, I have to `--soft-clip=true ... --convert-soft-clip-to-hard-clip=true`, to get _hard_ clipping, even though my command line explicitly says soft clipping.